src/UI/Component/Signal.php

```diff
@@ -12,4 +12,18 @@
  */
 interface Signal {
 
+	/**
+	 * Get the ID of this signal
+	 *
+	 * @return string
+	 */
+	public function getId();
```

Comment:
I know we discussed this, but I do not get at the moment, why this can not happen internally. Please summarize shortly again here.